### PR TITLE
Global pause feature

### PR DESCRIPTION
**Pull request by smalluban**
## Overview

Adds global pause feature. The feature uses a special ID to `options.paused` with 24 hours delay. 

To disable Ghostery completely it turns off the following features:
* Disables all DNR rulesets
* Disables WebRequest reporter (anti-tracking)
* Disables adblocker engines

## Wording & UI

* Added `Main protection` header (instead of `Protection modules`) - it fits more with the `Further protection`
* Updated privacy UI with a toggle
* Panel has new state - global pause - the main "trust/revoke" button has new action when the global pause is activated
* Added temporary notification about the feature - I recommend removing it in the next release - so it will be kept for at least a week on production

chrome-extension___jccobfekniccmgppgbkonmdmdldhhgjc_pages_settings_index html(iPad Air)
chrome-extension___jccobfekniccmgppgbkonmdmdldhhgjc_pages_settings_index html(iPad Air) (1)
Zrzut ekranu 2024-06-10 o 16 10 14

### CODE REVIEW

**Comment by philipp-classen** [comment]
Since we are doing linear scan over the `paused` array, I assume we are confident that it does not grow. Let's say, it stays under 50 entries or so, right?

Otherwise, we should consider switching to a set when holding it in memory.

**Comment by mdsandu** [comment]
- I would remove all titles from the settings like Main protection, Further protection - the separator lines are already stating the  different set up area 
- Pause in Global pause needs to be capitalized to be inline with the rest 
- there is an "i" in 23h 5 mins left - should that be "&", I would have no icons there

File: extension-manifest-v3/src/background/reporting/webrequest-reporter.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ if (__PLATFORM__ !== 'safari') {
   });
 
   observe('paused', (paused) => {
-    pausedHostnames = paused || [];
+    pausedHostnames = paused.map(({ id }) => id) || [];
```
[the remark this answers is not in the document]
Comment (smalluban) [reply extension-manifest-v3/src/background/reporting/webrequest-reporter.js:43]: Some time ago Safari did not have `options.paused` field, but it is not valid anymore.

**Comment by smalluban** [comment]
> there is an "i" in 23h 5 mins left - should that be "&", I would have no icons there

@mdsandu This is a "polish" artifact, I removed polish from translations, but this time counter is generated by the browser, and it used polish internally, it will be different in English, sorry for misleading you.

**Comment by smalluban** [comment]
chrome-extension___jccobfekniccmgppgbkonmdmdldhhgjc_pages_settings_index html(iPad Air) (2)

Tweaked view @mdsandu (full English)

**Comment by smalluban** [comment]
@philipp-classen The `options.paused` is a list created by the user by hand - it might have more than 50 entries, but I suppose it rarely changes.

**Comment by mdsandu** [comment]
- [ ] Title: Global Pause —> Pause Ghostery

Desc:  Suspends privacy protection globally for 1 day.
Screenshot 2024-06-11 at 12 19 06


- [ ] Panel - Bottom text:

Title (bold if possible): Missing the “Pause Ghostery” feature? 
Description (force new line): You can find it in your Privacy protection settings.

CTA: Open Ghostery settings 
Screenshot 2024-06-11 at 12 19 22

related - same page:

- [ ] Never-Consent
Automatically rejects of cookie consent notices. --> remove "of" 
Screenshot 2024-06-11 at 12 19 50

- [ ]  Search Engine Redirect Protection
Prevent Google from redirecting search result links through their servers instead of linking directly to pages. --> change to "Prevents" 
Screenshot 2024-06-11 at 12 20 43

**Review by chrmod — APPROVED** [review]

**Pull request merged**